src/renderer/components/screens/AccountListSidebar/AccountListSidebar.tsx

```diff
@@ -0,0 +1,119 @@
+import { T } from '@deltachat/jsonrpc-client'
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import { BackendRemote } from '../../../backend-com'
+import { AccountItem } from './AccountItem'
+
+import styles from './styles.module.scss'
+import { AccountHoverInfo } from './AccountHoverInfo'
+import { debounce } from 'debounce'
+import { runtime } from '../../../runtime'
+
+export function AccountListSidebar({
+  selectedAccountId,
+  onAddAccount,
+  onSelectAccount,
+  openAccountDeletionScreen,
+}: {
+  selectedAccountId: number | undefined
+  onAddAccount: () => Promise<number>
+  onSelectAccount: (accountId: number) => Promise<void>
+  openAccountDeletionScreen: (accountId: number) => Promise<void>
+}) {
+  const [accounts, setAccounts] = useState<T.Account[]>([])
+
+  const selectAccount = async (accountId: number) => {
+    if (selectedAccountId === accountId) {
+      return
+    }
+
+    await onSelectAccount(accountId)
+  }
+
+  const [syncAllAccounts, setSyncAllAccounts] = useState(true)
+
+  const refresh = useMemo(
+    () => async () => {
+      const accounts = await BackendRemote.rpc.getAllAccounts()
+      setAccounts(accounts)
+      const desktopSettings = await runtime.getDesktopSettings()
+      setSyncAllAccounts(desktopSettings.syncAllAccounts)
+    },
+    []
+  )
```

Comment:
This all smells a little bit like logic which might live well in a context rather than in the view component itself ("state / logic required to manage the account sidebar")

Reply:
desktop settings are only used for the sync state and that will be stored in core in the future for each account separately so the info will become integrated into the result of `BackendRemote.rpc.getAllAccounts`.

Though the window variable hack for triggering to update it might live better in a react context on app level? or it's own store? so it can be triggered wherever it is needed.
The actual result is only needed here and passed down to the `AccountItem` components, which are nearly pure components, the only thing not pure is the unread count (and if you are pedantic also the `useTranslationFunction` hook)

Reply:
Ideally when we have the core events it will just a simple use effect like updating unread state in the accounts:
```ts
  const [unreadCount, setUnreadCount] = useState<number>(0)
  useEffect(() => {
    const update = debounce(() => {
      BackendRemote.rpc
        .getFreshMsgs(account.id)
        .then(u => setUnreadCount(u?.length || 0))
        .catch(log.error)
    }, 200)
    update()
    const cleanup = [
      onDCEvent(account.id, 'IncomingMsg', update),
      onDCEvent(account.id, 'MsgsNoticed', update),
    ]
    return () => cleanup.forEach(off => off())
  }, [account.id])
```
maybe a bit more complex to handle updating single accounts instead of all of them if we want that optimisation